apiserver/cmd/apiserver/server/watch.go

```diff
@@ -30,10 +29,16 @@ func WatchExtensionAuth(stopChan chan struct{}) (bool, error) {
 	// set up k8s client
 	// attempt 1: KUBECONFIG env var
 	cfgFile := os.Getenv("KUBECONFIG")
-	cfg, err := clientcmd.BuildConfigFromFlags("", cfgFile)
+	// Host env vars bleed through to the container on Windows HPC, so if cannot
+	// be trusted in this case
+	// FIXME: this will no longer be needed when containerd v1.6 is EOL'd
+	if winutils.InHostProcessContainer() {
+		cfgFile = ""
+	}
+	cfg, err := winutils.BuildConfigFromFlags("", cfgFile)
 	if err != nil {
 		// attempt 2: in cluster config
-		if cfg, err = rest.InClusterConfig(); err != nil {
+		if cfg, err = winutils.GetInClusterConfig(); err != nil {
```

Comment:
I've noticed this takes the place of our old function for retrieving the in cluster config. Is this general-purpose enough where it can replace our old call for Linux envs too?

Reply:
Yes it is, `winutils.GetInClusterConfig()` is an exact copy of `rest.InClusterConfig()` except for it uses `winutils.GetHostPath()` on the `tokenFile` and `rootCAFile`. `winutils.GetHostPath()` in turn, is a no-op on Linux envs (and on non-HPC windows envs too). I've intentionally left `rest.InClusterConfig()` in tests so that we're able to catch it if the functions drift apart. Hopefully containerd v1.7 will be widely adopted soon and we can drop these workarounds... Thanks @mgleung!